clang/test/Modules/lambda-merge.cpp

```diff
@@ -1,4 +1,5 @@
 // RUN: %clang_cc1 -fmodules -std=c++17 -emit-llvm %s -o - -triple x86_64-linux-gnu | FileCheck %s
+// RUN: %clang_cc1 -fmodules -std=c++17 -emit-llvm %s -o - -triple x86_64-linux-gnu -fexperimental-new-constant-interpreter | FileCheck %s
```

Comment:
Is this fixing the fact that the test below would not pass w/o this change?

Reply:
Yep, apparently a declaration is being merged after the definition so we used to try to compile the declaration.